### PR TITLE
JWT-Auth upgrade to 2.1 version

### LINKED ISSUE #5819: 4.x: Upgrade MP JWT-Auth to 2.1

### PR DESCRIPTION
**Pull request by Verdent**
Fixes #5819

### CODE REVIEW

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (spericas) [inline microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:120]: Addition is good, but it would be nice to order public before private static fields.

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (spericas) [inline microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:1255]: Required param non-null?

File: security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java
Comment (spericas) [inline security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java:208]: Could this be an OR conditional?

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (Verdent) [reply microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:120]: Good point, I will change that. :-)

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (Verdent) [reply microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:1255]: This can be null. I think better would be probably to change the wording to not say required, but expected :-) This required could be a bit misleading.

File: security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java
Comment (Verdent) [reply security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java:208]: Yep, make sense.

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (Verdent) [reply microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:120]: Hmmm I took a look into this and right now and public static fields are before the private ones. You are right, there is LOGGER and JSON defined before that, but those should not be moved from my point of view. I would very much rather keep it the way it is now, since public property names are before the private ones already. Does it make sense?

**Comment by danielkec** [comment]
Dependency change needs to be reviewed by release manager @barchetta

File: security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java
Comment (danielkec) [inline security/jwt/src/main/java/io/helidon/security/jwt/EncryptedJwt.java:177]: :space_invader:

File: microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java
Comment (Verdent) [reply microprofile/jwt-auth/src/main/java/io/helidon/microprofile/jwt/auth/JwtAuthProvider.java:118]: changed.

**Review by danielkec — APPROVED** [review]
LGTM please wait for @barchetta review

**Pull request merged**
